synapse/rest/client/v1/login.py

```diff
@@ -207,6 +251,8 @@ def _do_other_login(self, login_submission):
                 "user": user_id,
             }
 
+            self._ratelimit_user(user_id)
```

Comment:
how about doing this inside the auth handler? that would mean:

* you wouldn't have to add the user_rate_limiter to `HomeServer` (you can just create it in the auth handler)
* you wouldn't have to special-case `m.id.user` below